weather/src/main/java/com/oddle/app/weather/adapter/OpenWeatherGatewayAdapter.java

```diff
@@ -0,0 +1,26 @@
+package com.oddle.app.weather.adapter;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import com.oddle.app.weather.config.OpenWeatherConfig;
+import com.oddle.app.weather.dto.response.openweather.OWCurrentWeatherResponse;
+
+@Component
+public class OpenWeatherGatewayAdapter {
+	
+	@Autowired
+	private OpenWeatherConfig openWeatherConfig;
+
+	public OWCurrentWeatherResponse getCurrentWeather(String city) {
+		RestTemplate restTemplate = new RestTemplate();
+		
+		UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(openWeatherConfig.getCurrentWeatherUri())
+		        .queryParam("q", city)
```

Comment:
we have problem trying to get weather for "Ho Chi Minh", can you let us know what is the problem here and how you will fix that? No need to commit new code

Reply:
on line 24, builder.toUriString() causes url encoding issue that the api gets incorrect parameter,

to solve that you can change to line 24 builder.toUriString() to builder.build().toUri()